api/src/wfl/executor.clj

```diff
@@ -428,6 +446,88 @@
      executor
      (jdbc/query tx [(format query details) status]))))
 
+(defn ^:private reference-id-to-workflow-records
```

Comment:
I do think this is worth it because the name was fuzzy for even me to grasp, and I'm the author.  You're right that I named based on the return type.

I'll look into this, may be a scenario where breaking out into an external method reduces clarity rather than improving it.

Reply:
I refactored as suggested but renamed thusly:

```
(defn ^:private workflow-and-sibling-records
  "Return the workflow records for all workflows in submissions
  associated with the specified `workflows` --
  i.e. records for `workflows` and their siblings."
  …
```

Because `workflows-to-workflow-records` implies that you're getting the workflow records for the specified workflows, and this is a funky case.